xfp/xdict.py

```diff
@@ -0,0 +1,361 @@
+        return cast(Xresult[IndexError, X], Xtry.from_unsafe(lambda: self.get(y)))
+
+    def set_item(self, key: Y, value: E) -> "Xdict[Y, X | E]":
+        """Return a new Xdict, with an updated couple (key: Y, value: E).
+
+        Upsert a new `value` at `key`.
+
+        ### Usage
+
+        ```python
+        from xfp import Xdict
+
+        assert Xdict({"a": 1}).set_item("b", 2) == Xdict({"a": 1, "b": 2})
+        assert Xdict({"a": 1}).set_item("a", 2) == Xdict({"a": 2})
+        ```
+        """
+        return self.union(Xdict({key: value}))
+
+    def del_item(self, key: Y) -> "Xdict[Y, X]":
+        """Return a new Xdict, with the given key deleted.
+
+        Filter the provided key if found.
+        No error is raised if the key doesn't exist.
+
+        ### Usage
+
+        ```python
+        from xfp import Xdict
+
+        assert Xdict({"a": 1, "b": 2}).del_item("b") == Xdict({"a": 1})
+        assert Xdict({"a": 1}).del_item("b") == Xdict({"a": 1})
+        ```
+        """
+        return self.filter(lambda y, _: y is not key)
+
+    def union(self, other: ABCDict[E, E]) -> "Xdict[Y | E, X | E]":
+        """Return a new Xdict, being the merge of self and a given one.
+
+        Works as if multiple set_items are done successively.
+        It means if a key is present in both Xdict, the `other` Xdict has priority.
+
+        ### Usage
+
+        ```python
+        from xfp import Xdict
+
+        assert Xdict({"a": 1, "b": 2}).union(Xdict({"a": 3, "c": 4})) == Xdict({"a": 3, "b": 2, "c": 4})
+        ```
+        """
+        return self.from_list(list(self.items()) + list(other.items()))
+
+    def keys(self) -> Xlist[Y]:
+        """Return an Xlist of the keys of the Xdict."""
+        return Xlist(self.__data.keys())
+
+    def values(self) -> Xlist[X]:
+        """Return an Xlist of the values of the Xdict."""
+        return Xlist(self.__data.values())
+
+    def items(self) -> Xlist[tuple[Y, X]]:
+        """Return an Xlist of the couples (key, value) of the Xdict."""
+        return Xlist(self.__data.items())
+
+    def map(self, f: Callable[[Y, X], tuple[E, E]]) -> "Xdict[E, E]":
+        """Return a new Xdict, after transformation of the couples (key, value) through `f`.
+
+        Transform each couple with `f`, then recreate an Xdict with the result.
+        In case of conflicts in the resulted key set, which value is associated with each key is an unenforced behavior,
+        meaning this case should be avoided.
+        However, consistency between executions is enforced and the same Xdict will be returned each time.
+
+        ### Usage
+
+        ```python
+        from xfp import Xdict
+
+        assert Xdict({"a": 1, "b": 2}).map(lambda y, x: (f"{y}{y}", x * 10)) == Xdict({"aa": 10, "bb": 20})
+
+        collisioned = Xdict({"a": 1, "b": 2}).map(lambda _, x: ("c", x * 10))
+        assert collisioned == Xdict({"c": 20}) or collisioned == Xdict({"c": 10}) # but it will always return the same
+        ```
```

Comment:
```suggestion
        ```python
            from xfp import Xdict

            assert Xdict({"a": 1, "b": 2}).map(lambda y, x: (f"{y}{y}", x * 10)) == Xdict({"aa": 10, "bb": 20})

            collisioned = Xdict({"a": 1, "b": 2}).map(lambda _, x: ("c", x * 10))
            assert collisioned == Xdict({"c": 20}) or collisioned == Xdict({"c": 10}) # but it will always return the same
        ```
```

Reply:
done